File: Difficulty: Easy/Anagram/anagram.py
#User function Template for python3
class Solution:
    
    # Function to check whether two strings are anagrams of each other.
    def areAnagrams(self, s1, s2):
        if len(s1) != len(s2):
            return False  # Quick check to avoid unnecessary work
        
        mp = {}
        for ch in s1:
            mp[ch] = mp.get(ch, 0) + 1  # Count characters in s1
        
        for ch in s2:
            if ch in mp:
                mp[ch] -= 1
                if mp[ch] < 0:
                    return False  # More occurrences in s2 than in s1
            else:
                return False  # Character not in s1 at all
        
        # If all values are zero, strings are anagrams
        return all(value == 0 for value in mp.values())

        #code here


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        a = input().strip()
        b = input().strip()
        if (Solution().areAnagrams(a, b)):
            print("true")
        else:
            print("false")
        print("~")

# } Driver Code Ends

File: Difficulty: Easy/Anagram/anagram.java
//{ Driver Code Starts
import java.io.*;
import java.lang.*;
import java.util.*;
import java.util.stream.*;

class GFG {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            String s1 = br.readLine(); // first string
            String s2 = br.readLine(); // second string

            Solution obj = new Solution();

            if (obj.areAnagrams(s1, s2)) {
                System.out.println("true");
            } else {
                System.out.println("false");
            }
            System.out.println("~");
        }
    }
}
// } Driver Code Ends


class Solution {
    // Function is to check whether two strings are anagram of each other or not.
    public static boolean areAnagrams(String s1, String s2) {
        if(s1.length()!=s2.length())
        {
            return false;
        }
        int[] charc=new int[128];
        for(int i=0;i<s1.length();i++)
        {
            charc[s1.charAt(i)]++;
        }
         for(int i=0;i<s1.length();i++)
        {
            charc[s2.charAt(i)]--;
        }
        for(int count:charc){
            if(count!=0)
            {
                return false;
            }
        }
        return true;
        
        // Your code here
    }
}